### PR TITLE
[Php80] Handle Call to a member function getOriginalValues() on string on AnnotationToAttributeRector

### PR DESCRIPTION
**Pull request by samsonasik**
This is to resolve https://github.com/rectorphp/rector/issues/6778

### CODE REVIEW

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/string_annotation_value.php.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;
+
+use Doctrine\ORM\Mapping as ORM;
+
+#[\Doctrine\ORM\Mapping\Table]
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/string_annotation_value.php.inc]: this possibly need to be : 

```diff
-#[\Doctrine\ORM\Mapping\Table]
+#[\Doctrine\ORM\Mapping\Table(name: App\Entity\Aktenzeichen::TABLE_NAME)]
```

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/string_annotation_value.php.inc]: Fixed 🎉 https://github.com/rectorphp/rector-src/pull/1099/commits/3bb2f7c87613f6b7bb2364eb940e3a0ceec37b17

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you 😌

**Comment by OskarStark** [comment]
@samsonasik I just tried the fix in a real project and there is a small detail missing.
A leading slash ❗ 

```diff
+#[\Doctrine\ORM\Mapping\Table(name: App\Entity\Aktenzeichen::TABLE_NAME)]
+#[\Doctrine\ORM\Mapping\Table(name: \App\Entity\Aktenzeichen::TABLE_NAME)]
```

Because the class itself is located in
```php
namespace App\Entity;
```

without the leading slash it tries to load the TABLE_NAME from:
```
App\Entity\App\Entity\Aktenzeichen::TABLE_NAME
```

which is wrong.

**Comment by samsonasik** [comment]
@OskarStark it get the string value, not handling string as class name/class const fetch in the arg, that mostly edge case? probably better to imported class name first before run it?

**Comment by OskarStark** [comment]
Importing the class won't work before in an annotation or am I wrong?

Any way we could use a regex to ensure it's a class constant ?

**Comment by samsonasik** [comment]
I think if it needs to be fully qualified, it should have a backslash in the first place on annotion before converted, or it will cause invalid assumption

**Comment by OskarStark** [comment]
I agree but right now annotations are always full qualified, because they does not take the current class name into account. So the behavior is different

**Pull request merged**